thirdeye-server/src/main/java/ai/startree/thirdeye/authorization/AccessControl.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2022 StarTree Inc
+ *
+ * Licensed under the StarTree Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at http://www.startree.ai/legal/startree-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT * WARRANTIES OF ANY KIND,
+ * either express or implied.
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package ai.startree.thirdeye.authorization;
+
+import ai.startree.thirdeye.auth.ThirdEyePrincipal;
+import java.net.http.HttpHeaders;
+
+public interface AccessControl {
```

Comment:
regarding packaging. I think now we have `ai.startree.thirdeye.auth` and `ai.startree.thirdeye.authorization`.

Not sure keeping it flat is the best idea.
- either we can have all authorization classes in auth as well. so auth = authentication + authorization
- else, if you foresee more classes coming in for authorization, then we can further the hierarchy like `ai.startree.thirdeye.auth.authorization`

Reply:
Ya, I wasn't too sure on this myself. I don't expect to need more authorization classes, so I'll put these in the auth pkg.